extensions/line-lengths-3.3.md

```diff
@@ -0,0 +1,86 @@
+---
+title: IRCv3.3 `maxline` Extension
+layout: spec
+copyrights:
+  -
+    name: "Daniel Oaks"
+    period: "2016"
+    email: "daniel@danieloaks.net"
+---
+A method for negotiating longer protocol lines.
+
+## Introduction
+
+Currently, IRC lines are limited to 512 bytes. There has been much desire to allow longer lines while sending or receiving IRC traffic, allowing longer lines with `PRIVMSG`/`NOTICE`, and solving related issues that come into play while working with the IRC protocol. These issues in particular come into play as the protocol is extended to provide greater functionality.
+
+The `maxline` capability specifies the maximum length of the tags, and of the rest of the message, that the server allows. As appropriate, lines will be truncated or split to account for clients which are still restricted to the standard (pre-maxline) limit.
+
+## Architecture
+
+### The `maxline` Capability
+
+The `maxline` capability, when advertised, MUST have a value which is a positive integer. For example:
+
+    C: CAP LS
+    S: CAP * LS :maxline=2048 
+
+Similarly to standard message handling, tags and the rest of the message have separate length values. The value of the `maxline` capability represents the maximum number of bytes that the tags section, and that the rest of the message, can take up. Line length calculation is done this way in order to better integrate with methods currently used by IRC software to limit line lengths.
```

Comment:
Done, thanks very much for the feedback! This also lets it work much more nicely with the new `message-tags` changes that are coming.